cmd/agent/DEVELOPERS.md

```diff
@@ -3,6 +3,18 @@
 This document provides relevant instructions for maintainers of the Grafana
 Agent.
 
+## Branching Strategy
+
+The Grafana Agent's branching strategy has shifted with the release of v0.19.0
+and is moving to a main always releasable approach. This will allow us to better control
```

Comment:
I don't think we have to mention when this changed, we can just talk about the approach & why